sklearn/tree/tree.py

```diff
@@ -297,9 +297,13 @@ def fit(self, X, y, sample_weight=None, check_input=True,
                 sample_weight = expanded_class_weight
 
         # Set min_weight_leaf from min_weight_fraction_leaf
-        if self.min_weight_fraction_leaf != 0. and sample_weight is not None:
+        if self.min_weight_fraction_leaf != 0.:
+            if sample_weight is None:
+                sample_weight = np.repeat(1., n_samples)
```

Comment:
Firstly, we don't need to explicitly do this, since all we want is the sum of weights, i.e. "total weight". So you could just do `min_weight_leaf = self.min_weight_fraction_leaf * n_samples`.

Secondly, I see why you might want this, but it now replicates the function of `min_samples_leaf`. So if this is the way we go, i.e. rather than a warning, you can actually just use
```python
min_samples_leaf = max(min_samples_leaf, int(ceil(self.min_weight_fraction_leaf * n_samples)))
```

In terms of testing, you should look at existing tests for `min_samples_leaf` and check that `min_weight_fraction_leaf` offers the same behaviour.